.travis.yml

```diff
@@ -10,14 +10,19 @@ cache:
   yarn: true
   directories:
     - "${HOME}/virtualenv/python$(python -c 'import platform; print(platform.python_version())')"
-    - '$NODE_DIR'
+    - '$VOLTA_HOME'
     - node_modules
     - '${HOME}/google-cloud-sdk'
 
 addons:
   apt:
     update: true
     packages:
+      # Xenial's libssl-dev provides an openssl 1.0 SO.
+      # Until we can upgrade the linux distribution, we're stuck here.
+      # Volta also ships binaries linked against openssl 1.1, but then we'd have to build it ourselves.
+      # There's also volta-cli/volta/pull/754 which builds volta with rustls to avoid this.
+      - libssl-dev
```

Comment:
Forgive my ignorance of the CI environment, but does Xenial not ship OpenSSL at all? And if it doesn't, does it not provide a non-dev package to install, instead of the development one? For _running_ Volta, you should just need the `.so` files, which I would expect to be installed by a non-dev package.

(For some reason I feel like I've had a similar discussion with @mitsuhiko before, but I can't find it in email now ☹️ )

Reply:
Xenial's `libssl-dev` provides the SO. https://packages.ubuntu.com/xenial/amd64/libssl-dev/filelist

Reply:
Oh, sorry, that's `libssl.so`.

Reply:
Right, volta (master) links against `libssl.so.1.1 => /usr/lib/x86_64-linux-gnu/libssl.so.1.1 (0x00007fa56d07f000)`

Reply:
There's a bit of nuance there: Building Volta from source on Linux will link against whatever version of OpenSSL is available on the machine doing the building (detected by the `openssl-sys` crate).

However, for the production builds, we compile and package two separate binaries, one linked against OpenSSL 1.0.x and the other against 1.1.x. So whichever version is available on the machine, we should have a binary that is compatible with it.

Reply:
If it's the case the Xenial doesn't come with OpenSSL at all by default, you should still be able to use the base package libssl1.0.0 (https://packages.ubuntu.com/xenial/amd64/libssl1.0.0/filelist) instead of the `-dev` package, which includes the headers for compiling against OpenSSL.

Reply:
TIL! Thanks a lot, that looks a lot better. libssl-dev was confusing to me, too.

Reply:
FWIW, it looks like Xenial should have the `libssl1.0.0` package by default: http://releases.ubuntu.com/xenial/ubuntu-16.04.6-desktop-amd64.manifest lists all the packages that are included in the base image.

Reply:
Yep, I'm testing a commit without any openssl packages, since it could have just been the case that 1.0.0 SO was there already. I didn't test that scenario yet.

Reply:
Lol Travis simply isn't triggering. Good stuff.

Reply:
Alright, it seems to be working without `libssl1.0.0`, presumably because Travis xenial has it. Thanks for the TILs!